### PR TITLE
Fixed belongsToMany relations eager loading issue

### PR DESCRIPTION
**Pull request by k3rnel**
## Description
Fixed belongsToMany relations eager loading issue

## How to test this
Try to eager load any existing belongsToMany relation

## Documentation
- [ ] My pull request requires a update on the documentation repository.

### CODE REVIEW

File: packages/Webkul/Product/src/Models/Product.php [hunk 1]
```diff
@@ -321,7 +321,7 @@ public function getEditableAttributes($group = null, $skipSuperAttribute = true)
     public function getAttribute($key)
     {
         if (! method_exists(static::class, $key)
-            && ! in_array($key, ['parent_id', 'attribute_family_id'])
+            && ! in_array($key, ['pivot', 'parent_id', 'attribute_family_id'])
```
Comment (devansh-webkul) [inline packages/Webkul/Product/src/Models/Product.php:324]: Hi, may I know the reason because I haven't seen any `pivot` column in the `products` table?

Comment (vivek-webkul) [reply packages/Webkul/Product/src/Models/Product.php:324]: Hi @devansh-webkul 
It's coming in the case of linked products (i.e. up_sells, cross_sells, and related_products)

error-related-products

Comment (k3rnel) [reply packages/Webkul/Product/src/Models/Product.php:324]: Hi. pivot its a name of 3rd table relation when you want to load belongsToMany relation

**Comment by devansh-webkul** [comment]
Hi @k3rnel,

I am checking this and will let you know if any changes are required.

**Comment by devansh-webkul** [comment]
Hi @k3rnel,

If you check the case that case is only for the `AttributeRepository`. If the key does not present then the parent method will always run. So eager loading will not cause any issue.

And I tested every `belongsToMany` relationship in the master version and it's working fine. The error which I have seen in the screenshot is due to a different reason.

Selection_001

**Comment by devansh-webkul** [comment]
Hi @k3rnel,

My mistake, I am doing lazy loading again :man_facepalming: With eager loading, it's showing the warning. And I tested this PR and it's working fine.

- Before

    Selection_002

- After

    Selection_003

Thanks for your contribution. :rocket:

**Review by devansh-webkul — APPROVED** [review]

**Pull request merged**